### PR TITLE
Replace old GIFs with symbols

### PR DESCRIPTION
**Pull request by strangelookingnerd**
Following a brief discussion with @MarkEWaite (https://github.com/jenkinsci/groovy-postbuild-plugin/pull/81#issuecomment-2404998761) I propose to get rid of the *.gif images.

In order to not break current pipelines an on-the-fly replacement should be implemented. See the following table for my proposal:

|name|old|new|origin|
|-|-|-|-|
|completed.gif|Image|grafik|core `status-blue`|
|db_in.gif|Image|grafik|ionicons `cloud-upload-outline`|
|db_out.gif|Image|grafik|ionicons `cloud-download-outline`|
|delete.gif|Image|grafik|core `trash`|
|error.gif|Image|grafik|core `status-red`|
|folder.gif|Image|grafik|core `folder`|
|green.gif|Image|grafik|emoji `green_square`|
|info.gif|Image|grafik|core `information-circle`|
|red.gif|Image|grafik|emoji `red_square`|
|save.gif|Image|grafik|ionicon `save-outline`|
|success.gif|Image|grafik|core `status-blue`|
|text.gif|Image|grafik|core `document-text`|
|warning.gif|Image|grafik|core `status-yellow`|
|yellow.gif|Image|grafik|emoji `yellow_square`|

Some of the icons do not have proper replacements from the core (https://github.com/jenkinsci/jenkins/tree/master/war/src/main/resources/images/symbols) or the `ionicons-api-plugin`. That's why I added `emoji-symbols-api` as a dependency to find the missing pieces.

Any feedback is highly welcome!

### Testing done

Added tests to cover the replacement

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/com/jenkinsci/plugins/badge/dsl/AddSummaryStep/help-icon.html [hunk 1]
```diff
@@ -27,9 +27,9 @@
         These icons are listed in the <a href="https://plugins.jenkins.io/badge/#plugin-content-icons">plugin documentation</a> and are referenced as "<i>image-name</i>.gif".
         For example, the following are all valid references to icons included with the plugin:
```
Comment (MarkEWaite) [inline src/main/resources/com/jenkinsci/plugins/badge/dsl/AddSummaryStep/help-icon.html:28]: The plugins in the webapps directory are provided by Jenkins, as far as I know.  Optional change that is unrelated to this pull request.
```suggestion
        For example, the following are all valid references to icons that can be used with the plugin:
```

**Review by MarkEWaite — APPROVED** [review]
I've tested each of the replacements in my Jenkins 2.479.1 controller and they look much better.  I've tested several of the gif images from the webapps directory and confirmed that they look good in the build history and continue to be visibly pixelated int he job summary page.

Pipeline that I used to test was:

```
pipeline {
    agent {
        label 'alpine'
    }
    stages {
        stage('Badge') {
            steps {
                addBadge icon: 'folder-open.gif', text: 'This is a folder open 16x16 icon'
                script {
                    manager.createSummary('folder-open.gif').appendText('This is a folder open')
                }
            }
        }
    }
}
```

**Review by basil — COMMENTED** [review]
```
org.junit.ComparisonFailure: expected:<[/plugin/badge/images/info.gif]> but was:<[symbol-information-circle]>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.jvnet.hudson.plugins.groovypostbuild.GroovyPostbuildRecorderTest.testBadgeMigration(GroovyPostbuildRecorderTest.java:481)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:706)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```
